src/os_posix/os_fs.c

```diff
@@ -676,17 +676,25 @@ __posix_open_file(WT_FILE_SYSTEM *file_system, WT_SESSION *wt_session,
 
 #if defined(HAVE_POSIX_FADVISE)
 	/*
-	 * Disable read-ahead on trees: it slows down random read workloads.
+	 * If the user set an access pattern hint, call fadvise now.
 	 * Ignore fadvise when doing direct I/O, the kernel cache isn't
 	 * interesting.
 	 */
-	if (!pfh->direct_io && file_type == WT_FS_OPEN_FILE_TYPE_DATA) {
+	if (!pfh->direct_io && file_type == WT_FS_OPEN_FILE_TYPE_DATA &&
+	    LF_ISSET(WT_FS_OPEN_ACCESS_RAND |
+	    LF_ISSET(WT_FS_OPEN_ACCESS_SEQ))) {
+		if (LF_ISSET(WT_FS_OPEN_ACCESS_RAND))
+			advise_flag = POSIX_FADV_RANDOM;
+		if (LF_ISSET(WT_FS_OPEN_ACCESS_SEQ))
+			advise_flag = POSIX_FADV_SEQUENTIAL;
```

Comment:
I'm surprised this doesn't generate an unused variable warning: how can the compiler figure out that `advise_flag` will always be set here?